src/Common/src/TypeSystem/Common/Utilities/GCPointerMap.cs

```diff
@@ -130,6 +130,25 @@ public override string ToString()
                 sb.Append(bit ? '1' : '0');
             return sb.ToString();
         }
+
+        public int CompareTo(object obj)
+        {
+            if (!(obj is GCPointerMap))
+                throw new ArgumentException();
+
+            GCPointerMap other = (GCPointerMap)obj;
+            if (_numCells != other._numCells)
+                return _numCells < other._numCells ? -1 : 1;
+
+            for (int i = 0; i < _gcFlags.Length; i++)
```

Comment:
You mean the remaining bits when the end of the encoding doesn't take up the full int? If so, that's guarded by the `_numCells` check. Any unused bits at the end should be zeroes and not be a problem for the comparison. I can switch to the `BitEnumerator` if you think that's more appropriate but comparing ints seemed faster (and hasn't shown any ordering instability in testing so far).